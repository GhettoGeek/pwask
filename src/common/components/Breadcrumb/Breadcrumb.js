import React from 'react'
import PropTypes from 'prop-types'
import { withRouter } from 'react-router-dom'
import { Breadcrumbs } from '@material-ui/core'
import { Home as HomeIcon } from '@material-ui/icons'
import { makeStyles } from '@material-ui/core/styles'
import Link from '../Link'
import Typography from '../Typography'

const useStyles = makeStyles({
  icon: {
    display: 'flex',
    width: 20,
    height: 20,
  },
})

const BreadCrumb = ({ location }) => {
  const classes = useStyles()
  const crumbs = location.pathname.split('/').filter(Boolean)

  return crumbs.length > 0 ? (
    <Breadcrumbs aria-label="Breadcrumb">
      <Link color="inherit" to="/">
        <HomeIcon className={classes.icon} />
      </Link>
      {
        crumbs.map((item, index) => {
          if (crumbs.length === index + 1) {
            return <Typography variant="body1" key={index}>{item}</Typography>
          }

          const link = crumbs.slice(0, index + 1).join('/')

          return <Link to={`/${link}`} key={index}>{item}</Link>
        })
      }
    </Breadcrumbs>
  ) : null
}

BreadCrumb.propTypes = {
  location: PropTypes.shape().isRequired,
}

export default withRouter(BreadCrumb)
