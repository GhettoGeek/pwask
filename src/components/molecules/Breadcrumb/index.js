import React from 'react'
import PropTypes from 'prop-types'
import { withRouter } from 'react-router-dom'
import Link from '@material-ui/core/Link'
import Typography from '@material-ui/core/Typography'
import Breadcrumbs from '@material-ui/lab/Breadcrumbs'
import HomeIcon from '@material-ui/icons/Home'
import { styled } from '../../../utils/style'

const StyledHomeIcon = styled(HomeIcon)(theme => ({
  marginRight: theme.spacing.unit / 2,
  width: 20,
  height: 20,
}))

const BreadCrumb = ({ location }) => {
  const crumbs = location.pathname.split('/').filter(Boolean)
  const crumbsLength = crumbs.length

  return (
    <Breadcrumbs aria-label="Breadcrumb">
      <Link color="inherit" href="/">
        <StyledHomeIcon />
      </Link>
      {
        crumbs.map((item, index) => {
          if (crumbsLength === index + 1) {
            return <Typography color="primary" key={index}>{item}</Typography>
          }

          return (
            <Link color="prmary" href={`/${item}`} key={index}>
              {item}
            </Link>
          )
        })
      }
    </Breadcrumbs>
  )
}

BreadCrumb.propTypes = {
  location: PropTypes.shape().isRequired,
}

export default withRouter(BreadCrumb)
