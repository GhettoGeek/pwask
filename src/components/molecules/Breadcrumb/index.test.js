import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'

import Breadcrumb from '.'

describe('Breadcrumb', () => {
  it('renders correctly the component', () => {
    const tree = renderer.create(
      <MemoryRouter initialEntries={['/']} initialIndex={0}>
        <Breadcrumb />
      </MemoryRouter>
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
