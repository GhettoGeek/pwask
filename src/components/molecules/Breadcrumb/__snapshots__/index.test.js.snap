// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb renders correctly the component 1`] = `
<nav
  aria-label="Breadcrumb"
  className="MuiTypography-root-5 MuiTypography-body1-14 MuiTypography-colorTextSecondary-38 MuiBreadcrumbs-root-1"
>
  <ol
    className="MuiBreadcrumbs-ol-2"
  >
    <li
      className="MuiBreadcrumbs-li-3"
    >
      <a
        className="MuiTypography-root-5 MuiTypography-colorInherit-34 MuiTypography-inline-40 MuiLink-root-41 MuiLink-underlineHover-43"
        href="/"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-47 StyledComponent-root-46"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
          />
          <path
            d="M0 0h24v24H0z"
            fill="none"
          />
        </svg>
      </a>
    </li>
  </ol>
</nav>
`;
